src/Search.php

```diff
@@ -5182,6 +5182,14 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
                                AND `$table`.`itemtype` = '$itemtype_val')";
                 }
                 break;
+
+            case "glpi_changes.content":
+            case "glpi_problems.content":
+            case "glpi_tickets.content":
+                if (in_array($searchtype, ['contains', 'notcontains'])) {
+                    $val = str_replace('&#38;', '38;amp;', $val);
+                }
+                break;
```

Comment:
I think a more global fix could be made. Indeed, if this transformation is moved in the `switch ($searchtype)` / `case "contains"`, with a `if (($searchopt[$ID]['datatype'] ?? null) === 'text') && ($searchopt[$ID]['htmltext'] ?? null) === true)` condition, it will fix this kind of search in all rich text fields (for instance, a search in followups, tasks, ...).